### PR TITLE
Defer z_getbalance and z_gettotalbalance disablement.

### PR DESCRIPTION
**Pull request by nuttycom**
These disable-by-default changes need to be deferred until we have a better replacement that provides a single source for summary balance information.

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -94,8 +94,6 @@ will be removed in 18 weeks:
 - `legacy_privacy`
 - `getnewaddress`
 - `getrawchangeaddress`
-- `z_getbalance`
-- `z_gettotalbalance`
```
Comment (str4d) [inline doc/release-notes.md:98]: `doc/book/src/user/deprecation.md` needs to be updated to logically revert the changes from https://github.com/zcash/zcash/commit/09cd65327b9b32b4992be020a71377f45eb0bdf6 (in practice, moving the two relevant lines from the Stage 2 table to the Stage 1 table.

File: src/deprecation.h [hunk 2]
```diff
@@ -27,6 +27,8 @@ static const std::set<std::string> DEFAULT_ALLOW_DEPRECATED{{
 
     // Wallet-level features
 #ifdef ENABLE_WALLET
+    ,"z_getbalance"
+    ,"z_gettotalbalance"
```
Comment (str4d) [inline src/deprecation.h:31]: I hate this syntax, but see why it's been done, and we'll be re-doing this change before `gbd_oldhashes` is disabled by default, so this is fine as-is.

File: doc/release-notes.md (hunk 1 above)
Comment (nuttycom) [reply doc/release-notes.md:98]: Thanks, fixed.

File: src/deprecation.h (hunk 2 above)
Comment (nuttycom) [reply src/deprecation.h:31]: The alternative is to put this block before `gbd_oldhashes` (and maybe make the same change in the order of node/wallet deprecation blocks below?) That would be resilient to future changes and wouldn't require the leading commas.

Comment (str4d) [reply src/deprecation.h:31]: It's only resilient while there are non-wallet features being deprecated. If we did that reorder, then the moment `gbd_oldhashes` is disabled by default, the line for the last of whatever wallet features were still enabled-by-default would need to be altered. This is just going to be a continual problem as long as the deprecations are encoded in C / C++ where we can't have trailing commas.

**Review by str4d — APPROVED** [review]
utACK 6d6b0137e10d9ab354783e51b76a9ece2d543b4b

File: src/deprecation.h (hunk 2 above)
Comment (nuttycom) [reply src/deprecation.h:31]: True. At least that way, though, all that would be required would be the removal of a single trailing comma. But we'll fix that the next time we touch this file.

Comment (daira) [reply src/deprecation.h:31]: I think C++11 onward *does* allow trailing commas in initializer lists ([first sentence of the abstract here](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2067r0.pdf)).

Comment (daira) [reply src/deprecation.h:31]: Building now to check whether that works in clang.

**Review by daira — APPROVED** [review]
utACK with suggestion.

File: src/deprecation.h (hunk 2 above)
Comment (daira) [reply src/deprecation.h:31]: It does.
```suggestion
    "z_getbalance",
    "z_gettotalbalance",
```
and a comma after `"gbt_oldhashes"` above (which is just outside the suggestion range).

**Review by sellout — APPROVED** [review]
utACK

**Review by daira — APPROVED** [review]
utACK c9d6230e88217315665a988ece7441eb371b6471

**Comment by daira** [comment]
CI failure:
```
Fetching utfcpp...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
[...]
curl: (28) Resolving timed out after 10000 milliseconds
Warning: Problem : timeout. Will retry in 1 seconds. 3 retries left.
[...]
curl: (28) Resolving timed out after 10000 milliseconds
make: *** [[funcs.mk:278](http://funcs.mk:278/): /workspace/c02fff64-21c4-45f4-9720-a8dc720ccae7/source/depends/sources/download-stamps/.stamp_fetched-utfcpp-utfcpp-3.2.3.tar.gz.hash] Error 28
make: Leaving directory '/workspace/c02fff64-21c4-45f4-9720-a8dc720ccae7/source/depends'
```
